test/Microsoft.VisualStudio.Editor.Razor.Test/DefaultVisualStudioRazorParserIntegrationTest.cs

```diff
@@ -463,106 +434,114 @@ private void ImplicitExpressionAcceptsIdentifierTypedAfterDotIfLastChangeWasProv
         }
 
-        // [Fact] Silent skip to avoid warnings. Skipping until we can control the parser more directly.
+        [ForegroundFact]
         private void ImplicitExpressionCorrectlyTriggersReparseIfUsingKeywordTyped()
         {
             RunTypeKeywordTest("using");
         }
 
-        // [Fact] Silent skip to avoid warnings. Skipping until we can control the parser more directly.
+        [ForegroundFact]
         private void ImplicitExpressionCorrectlyTriggersReparseIfSectionKeywordTyped()
         {
             RunTypeKeywordTest("section");
         }
 
-        // [Fact] Silent skip to avoid warnings. Skipping until we can control the parser more directly.
+        [ForegroundFact]
         private void ImplicitExpressionCorrectlyTriggersReparseIfInheritsKeywordTyped()
         {
             RunTypeKeywordTest("inherits");
         }
 
-        // [Fact] Silent skip to avoid warnings. Skipping until we can control the parser more directly.
+        [ForegroundFact]
         private void ImplicitExpressionCorrectlyTriggersReparseIfFunctionsKeywordTyped()
         {
             RunTypeKeywordTest("functions");
         }
 
-        // [Fact] Silent skip to avoid warnings. Skipping until we can control the parser more directly.
+        [ForegroundFact]
         private void ImplicitExpressionCorrectlyTriggersReparseIfNamespaceKeywordTyped()
         {
             RunTypeKeywordTest("namespace");
         }
 
-        // [Fact] Silent skip to avoid warnings. Skipping until we can control the parser more directly.
+        [ForegroundFact]
         private void ImplicitExpressionCorrectlyTriggersReparseIfClassKeywordTyped()
         {
             RunTypeKeywordTest("class");
         }
 
-        private TestParserManager CreateParserManager(ITextSnapshot originalSnapshot, int idleDelay = 50)
+        private TestParserManager CreateParserManager(ITextSnapshot originalSnapshot)
         {
-            var parser = new VisualStudioRazorParser(
-                Dispatcher, 
-                new TestTextBuffer(originalSnapshot), 
-                CreateTemplateEngine(), 
-                TestLinePragmaFileName,
+            var textBuffer = new TestTextBuffer(originalSnapshot);
+            var documentTracker = CreateDocumentTracker(textBuffer);
+            var templateEngineFactory = CreateTemplateEngineFactory();
+            var parser = new DefaultVisualStudioRazorParser(
+                Dispatcher,
+                documentTracker,
+                templateEngineFactory,
                 new DefaultErrorReporter(),
-                new TestCompletionBroker(), 
-                new Mock<VisualStudioDocumentTrackerFactory>().Object,                
-                new Mock<IEditorOperationsFactoryService>().Object);
+                new TestCompletionBroker(),
+                Enumerable.Empty<ICanHasContextChangedListener>())
+            {
+                IdleDelay = TimeSpan.FromMilliseconds(1),
```

Comment:
What's the impact of this for testing? Just a small value so it's fast but the timer is still involved?

Reply:
(add a comment)